runtime/drivers/file/watcher.go

```diff
@@ -175,27 +194,20 @@ func (w *watcher) runInner() error {
 
 			path = filepath.Join("/", path)
 			we.Path = path
+			we.FullPath = e.Name
 
 			// Do not send files for ignored paths
 			if drivers.IsIgnored(path, w.ignorePaths) {
 				continue
 			}
 
-			if e.Has(fsnotify.Create) {
-				info, err := os.Stat(e.Name)
-				we.Dir = err == nil && info.IsDir()
+			existing, ok := w.buffer[path]
+			if ok && existing.IsCreate {
+				// copy over `IsCreate` within the batch for a path
+				we.IsCreate = existing.IsCreate
 			}
```

Comment:
What if the file is rapidly created and deleted? Putting `IsCreate` on a delete event seems odd.

Reply:
The issue was older events overriding `Dir`. But calling `addDir` multiple times would be suboptimal. That is why we are doing it this way.

Reply:
Ya ignoring DELETE events. Newer delete events should overwrite older events.